### PR TITLE
Add Script to Available Domains

### PR DESCRIPTION
**Pull request by boojew**
This is a super minor mod to allow the import of scripts and treat them as scenes. This addresses my issue, #223

### CODE REVIEW

**Comment by schmittx** [comment]
FYI, some scripts can be cancelled if they include more than one service call, they'll include an attribute `can_cancel` set to `true` in that case.

It'd be nice to include support to turn off those supported scripts as well.

**Comment by boojew** [comment]
I did not know that. I'll look into how that works - all my scripts are fairly simple today.. Basically just enhanced scenes... but even then, they could probably be rewritten as scenes if I had some patience ;) 

So the idea would be that cancel would be triggered by turning the switch off? Not sure how that would really work as since I am importing the entire script as 1 scene - how would you cancel a specific service? Or am I missing some thing really dumb?

**Comment by schmittx** [comment]
A script is really just a list of service calls so when you call the `turn_off` service against that script it will stop the script wherever it is at in the list.

**Comment by boojew** [comment]
@schmittx I see.. hmm. ok, well, I can take a look at that, but likely won't be for a while, both due to time issues and limitted python skills.

**Comment by boojew** [comment]
@schmittx ok, after looking at this, I do not think I would be correctly make your suggestion work. I do think it is possible, but I dont see how to expose that properly. Can we merge this change as is?

**Comment by boojew** [comment]
Can we merge this?

File: accessories/switch.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ HomeAssistantSwitch.prototype = {
       this.log(`Setting power state on the '${this.name}' to on`);
 
       this.client.callService(callDomain, 'turn_on', serviceData, (data) => {
-        if (this.domain === 'scene') {
+        if (this.domain === 'scene' || this.domain === 'script') {
```
Comment (schmittx) [inline accessories/switch.js]: Replace line 62 with
`if (this.domain === 'scene' || (this.domain === 'script' && !(this.data.attributes.can_cancel))) {`

Then, replace line 171 with
`if (this.domain === 'remote' || this.domain === 'switch' || this.domain === 'input_boolean' || this.domain === 'group' || this.domain === 'automation' || (this.domain === 'script' && this.data.attributes.can_cancel)) {`

Comment (schmittx) [reply accessories/switch.js]: Also, please update the README to explain script support

I created a branch for script support to test my comments above, you can refer to it if you want (https://github.com/schmittx/homebridge-homeassistant/tree/script-support)

**Comment by robbiet480** [comment]
Thanks for the submission @boojew but I'm going to go with @schmittx's as it has the comments he suggested. :cookie:

**Pull request closed without merging**